<a href="https://colab.research.google.com/github/IBREEZZ/Code_Academy_Makeen2/blob/main/Exploring_Daily_Minimum_Temperatures_Dataset.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd

**Load the Dataset**

In [ ]:
df = pd.read_csv("daily-minimum-temperatures-in-me.csv")

In [ ]:
df.head()


Date Daily minimum temperatures
0  1/1/1981                       20.7
1  1/2/1981                       17.9
2  1/3/1981                       18.8
3  1/4/1981                       14.6
4  1/5/1981                       15.8

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3650 entries, 0 to 3649
Data columns (total 2 columns):
 #   Column                      Non-Null Count  Dtype 
---  ------                      --------------  ----- 
 0   Date                        3650 non-null   object
 1   Daily minimum temperatures  3650 non-null   object
dtypes: object(2)
memory usage: 57.2+ KB


In [ ]:
df.describe()

Date Daily minimum temperatures
count         3650                       3650
unique        3650                        230
top     12/31/1990                         10
freq             1                         51

**Convert Date Column**

In [ ]:
df['Date'] = pd.to_datetime(df['Date'])

In [ ]:
print(df.dtypes)

Date                          datetime64[ns]
Daily minimum temperatures            object
dtype: object


**Extract Features**

In [ ]:
df['Year'] = df['Date'].dt.year
df['Month'] = df['Date'].dt.month
df['Day'] = df['Date'].dt.day
df['Day_of_Week'] = df['Date'].dt.day_name()

In [ ]:
df.head()

Date Daily minimum temperatures  Year  Month  Day Day_of_Week
0 1981-01-01                       20.7  1981      1    1    Thursday
1 1981-01-02                       17.9  1981      1    2      Friday
2 1981-01-03                       18.8  1981      1    3    Saturday
3 1981-01-04                       14.6  1981      1    4      Sunday
4 1981-01-05                       15.8  1981      1    5      Monday

**Filter Data**

In [ ]:
df_1985 = df[df['Year'] == 1985]
df_1985

Date Daily minimum temperatures  Year  Month  Day Day_of_Week
1460 1985-01-01                       13.3  1985      1    1     Tuesday
1461 1985-01-02                       15.2  1985      1    2   Wednesday
1462 1985-01-03                       13.1  1985      1    3    Thursday
1463 1985-01-04                       12.7  1985      1    4      Friday
1464 1985-01-05                       14.6  1985      1    5    Saturday
...         ...                        ...   ...    ...  ...         ...
1820 1985-12-27                       11.5  1985     12   27      Friday
1821 1985-12-28                       10.8  1985     12   28    Saturday
1822 1985-12-29                         12  1985     12   29      Sunday
1823 1985-12-30                       16.3  1985     12   30      Monday
1824 1985-12-31                       14.4  1985     12   31     Tuesday

[365 rows x 6 columns]

In [ ]:
df_1985['Daily minimum temperatures'] = pd.to_numeric(df_1985['Daily minimum temperatures'], errors='coerce')


<ipython-input-29-a244f0a81077>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_1985['Daily minimum temperatures'] = pd.to_numeric(df_1985['Daily minimum temperatures'], errors='coerce')


In [ ]:
below_10_count = (df_1985['Daily minimum temperatures'] < 10).sum()
print(f"Number of days in 1985 with temperature below 10°C: {below_10_count}")

Number of days in 1985 with temperature below 10°C: 142


**Analyze Time Periods**

In [ ]:
df.rename(columns={'Daily minimum temperatures': 'Temp'}, inplace=True)

df['Temp'] = pd.to_numeric(df['Temp'], errors='coerce')


In [ ]:

yearly_avg_temp = df.groupby('Year')['Temp'].mean()

yearly_avg_temp


Year
1981    11.517260
1982    10.840220
1983    11.187397
1984    10.620604
1985    11.137534
1986    10.803288
1987    10.853151
1988    11.972055
1989    11.261918
1990    11.669589
Name: Temp, dtype: float64

In [ ]:
lowest_avg_year = yearly_avg_temp.idxmin()

print(f"Year with lowest avg minimum temp: {lowest_avg_year} ({yearly_avg_temp[lowest_avg_year]:.2f}°C)")


Year with lowest avg minimum temp: 1984 (10.62°C)


**Summary**

Based on my analysis of the Daily Minimum Temperatures dataset, I was able to successfully load and analyze the data, converting the date column and extracting time-based features like year, month, and day of the week. In 1985, there were 142 days with temperatures below 10°C, showing typical winter patterns. According to my calculations, the lowest average minimum temperature was 10.62°C in 1984.